pkg/operator/console.go

```diff
@@ -0,0 +1,35 @@
+package operator
+
+import (
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+
+	"github.com/openshift/console-operator/pkg/apis/console/v1alpha1"
+)
+
+func defaultConsole() *v1alpha1.Console {
+	console := &v1alpha1.Console{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "console.openshift.io/v1alpha1",
+			Kind:       "Console",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OpenshiftConsoleNamespace,
+			Namespace: OpenshiftConsoleNamespace,
+		},
+	}
+	console.SetDefaults()
+	return console
+}
+
+func ApplyConsole() (*v1alpha1.Console, error) {
+	console := defaultConsole()
+	if err := sdk.Get(console); errors.IsNotFound(err) {
+		if err = sdk.Create(console); err != nil {
+			return nil, err
+		}
+	}
+	return console, nil
```

Comment:
I'm not familiar with the SDK... Will this return the right value if the console already exists?

Reply:
Flow here is:
- `defaultConsole()` provides enough to make a `Get` request
- if got, it mutates `console` (dislike this about the api) and returns it.
- if not, it `Create`'s a `console`, still mutates `console`, and returns it.

I think the mutation is confusing and dislike that about the API.